### PR TITLE
Clean up all the things

### PR DESCRIPTION
**Pull request by kashizui**
moving towards PEP8 conformance
cleaning up bad/faulty/dangerous logic/practice
optimizing readability/maintainability

all tests pass

@percyliang @pbhatnagar3 please review

### CODE REVIEW

File: codalab/client/local_bundle_client.py [hunk 1]
```diff
@@ -34,7 +33,7 @@
 from codalab.objects.worksheet import Worksheet
 from codalab.objects import permission
 from codalab.objects.permission import (
-    check_bundles_have_read_permission, # unused
+    check_bundles_have_read_permission,  # unused
```
Comment (percyliang) [inline codalab/client/local_bundle_client.py]: why include if unused?

File: codalab/lib/formatting.py [hunk 2]
```diff
@@ -126,42 +133,34 @@ def parse_duration(s):
...
+    Converts string to list of tokens, with support for quotes.
+    Defined here for convenience.
+
+    See shlex documentation for more information on parsing rules:
+    https://docs.python.org/2/library/shlex.html#parsing-rules
+
+    :param s: string "a b 'c d' e"
+    :return: list ["a", "b", "c d", "e"]
+    """
+    return shlex.split(s, comments=False, posix=True)
```
Comment (percyliang) [inline codalab/lib/formatting.py:166]: Nice - good to know about this!

**Comment by percyliang** [comment]
Thanks for doing this - LGTM!
(Btw, did you do this manually or semi-automatically?)

**Comment by kashizui** [comment]
Semi-automatic, depending on the type of change. (a lot of formatting changes were automatic)

File: codalab/client/local_bundle_client.py (hunk 1 above)
Comment (kashizui) [inline codalab/client/local_bundle_client.py]: oops, not actually unused
removing comment

**Pull request merged**